test/e2e/case18_compliance_api_test.go

```diff
@@ -412,7 +499,372 @@ var _ = Describe("Test the compliance events API", Label("compliance-events-api"
+				[]float64{1, 2, 3},
+			),
+			Entry(
+				"Sort descending by policy.id",
+				[]string{"sort=policy.id", "direction=desc"},
+				[]float64{3, 2, 1},
+			),
+			Entry(
+				"Sort ascending by policy.id",
+				[]string{"sort=policy.id", "direction=asc"},
+				[]float64{1, 2, 3},
+			),
+			Entry(
+				"Sort descending by policy.kind",
+				[]string{"sort=policy.kind", "direction=desc"},
+				[]float64{1, 2, 3},
+			),
+			Entry(
+				"Sort ascending by policy.kind",
+				[]string{"sort=policy.kind", "direction=asc"},
+				[]float64{1, 2, 3},
+			),
+			Entry(
+				"Sort descending by policy.name",
+				[]string{"sort=policy.name", "direction=desc"},
+				[]float64{2, 3, 1},
+			),
+			Entry(
+				"Sort ascending by policy.name",
+				[]string{"sort=policy.name", "direction=asc"},
+				[]float64{1, 2, 3},
+			),
+			Entry(
+				"Sort descending by policy.namespace",
+				[]string{"sort=policy.namespace", "direction=desc"},
+				[]float64{2, 3, 1},
+			),
+			Entry(
+				"Sort ascending by policy.namespace",
+				[]string{"sort=policy.namespace", "direction=asc"},
+				[]float64{1, 2, 3},
+			),
+			Entry(
+				"Sort descending by policy.severity",
+				[]string{"sort=policy.severity", "direction=desc"},
+				[]float64{2, 3, 1},
+			),
+			Entry(
+				"Sort ascending by policy.severity",
+				[]string{"sort=policy.severity", "direction=asc"},
+				[]float64{1, 2, 3},
+			),
+			Entry(
+				"Sort descending by parent_policy.id and policy.id",
+				[]string{"sort=parent_policy.id,policy.id", "direction=asc"},
+				[]float64{1, 2, 3},
+			),
+			Entry(
+				"Sort descending by id",
+				[]string{"sort=id", "direction=desc"},
+				[]float64{3, 2, 1},
+			),
+			Entry(
+				"Sort ascending by id",
+				[]string{"sort=id", "direction=asc"},
+				[]float64{1, 2, 3},
+			),
+		)
+
+		Describe("Invalid event ID", func() {
+			It("Compliance event is not found", func(ctx context.Context) {
+				req, err := http.NewRequestWithContext(ctx, http.MethodGet, eventsEndpoint+"/1231291", nil)
+				Expect(err).ToNot(HaveOccurred())
+
+				resp, err := httpClient.Do(req)
+				Expect(err).ToNot(HaveOccurred())
+
+				defer resp.Body.Close()
+
+				Expect(resp.StatusCode).To(Equal(http.StatusNotFound))
+
+				body, err := io.ReadAll(resp.Body)
+				Expect(err).ToNot(HaveOccurred())
+
+				respJSON := map[string]any{}
+
+				err = json.Unmarshal(body, &respJSON)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(respJSON["message"].(string)).To(Equal("The requested compliance event was not found"))
+			})
+
+			It("Compliance event ID is invalid", func(ctx context.Context) {
+				req, err := http.NewRequestWithContext(ctx, http.MethodGet, eventsEndpoint+"/sql-injections-lose", nil)
+				Expect(err).ToNot(HaveOccurred())
+
+				resp, err := httpClient.Do(req)
+				Expect(err).ToNot(HaveOccurred())
+
+				defer resp.Body.Close()
+
+				Expect(resp.StatusCode).To(Equal(http.StatusBadRequest))
+
+				body, err := io.ReadAll(resp.Body)
```

Comment:
It doesn't really matter but I found https://daryl-ng.medium.com/why-you-should-avoid-ioutil-readall-in-go-e6be4de180f8